### PR TITLE
Add registration

### PR DESCRIPTION
**Pull request by GeorgePuisha**
Registration & signing in with [Auth0](https://auth0.com/#/). 

Still no adding users to DB. 

PR is provided with tests.

### CODE REVIEW

File: src/app/services/auth/auth.service.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import { Injectable } from "@angular/core";
+import { Router } from "@angular/router";
+import "rxjs/add/operator/filter";
+import * as auth0 from "auth0-js";
+
+@Injectable()
+export class AuthService {
+
+  auth0 = new auth0.WebAuth({
+    clientID: "BAEUD0UBR83uSbxpNJhHuLuGRjg2KEC5",
```
Comment (AndreiPiatrou) [inline src/app/services/auth/auth.service.ts]: Token and IDs should be extracted into one config file.

File: src/app/app.component.ts [hunk 2]
```diff
@@ -4,6 +4,8 @@ import { NavbarComponent } from "./components/navbar/navbar.component";
 import { FooterComponent } from "./components/footer/footer.component";
 import { GreetComponent } from "./components/greet/greet.component";
 
+import { AuthService } from "./services/auth/auth.service";
+
```
Comment (AndreiPiatrou) [inline src/app/app.component.ts]: Useless line.

File: src/app/services/auth/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        this.router.navigate(["/"]);
+      } else if (err) {
+        this.router.navigate(["/"]);
+      }
+    });
+  }
+
+  private setSession(authResult): void {
+    const expiresAt = JSON.stringify((authResult.expiresIn * 1000) + new Date().getTime());
+    console.log(expiresAt);
```
Comment (AndreiPiatrou) [inline src/app/services/auth/auth.service.ts]: Do not leave `console.log` in production code.

File: src/app/services/auth/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        this.setSession(authResult);
+        this.router.navigate(["/"]);
+      } else if (err) {
+        this.router.navigate(["/"]);
+      }
+    });
+  }
+
+  private setSession(authResult): void {
+    const expiresAt = JSON.stringify((authResult.expiresIn * 1000) + new Date().getTime());
```
Comment (AndreiPiatrou) [inline src/app/services/auth/auth.service.ts]: `1000` - hardcoded magic number.

File: src/app/app.component.ts (hunk 2 above)
Comment (GeorgePuisha) [reply src/app/app.component.ts]: I deny. Auth service is used in `app.component.ts`  to handle authentication. 

See `auth.service.ts`, `handleAuthentication()` function.

File: src/app/services/auth/auth.service.ts (hunk 3 above)
Comment (GeorgePuisha) [reply src/app/services/auth/auth.service.ts]: Fixed.

File: src/app/services/auth/auth.service.ts (hunk 4 above)
Comment (GeorgePuisha) [reply src/app/services/auth/auth.service.ts]: Fixed.

File: src/app/services/auth/auth.service.ts (hunk 1 above)
Comment (GeorgePuisha) [reply src/app/services/auth/auth.service.ts]: Fixed (as far as I see).

**Pull request merged**